app/assets/javascripts/discourse/app/mapping-router.js

```diff
@@ -134,12 +134,21 @@ export function mapRoutes() {
     }
   });
 
-  return BareRouter.extend({
+  const extendedRouter = BareRouter.extend({
     rootURL: getURL("/"),
   }).map(function () {
     tree.mapRoutes(this);
     this.route("unknown", { path: "*path" });
   });
+
+  _extendedRouter = extendedRouter;
+  return extendedRouter;
+}
+
+let _extendedRouter;
```

Comment:
Would it be possible to avoid storing the router in a global var and instead get the current router instance from the container? It should be available under `router:main` registration

Reply:
I tried at the beginning but couldn't have what I wanted... but maybe it was just my limited understanding at that point. Let me check again :+1: